src/main/java/app/service/impl/FileWriterServiceImplementation.java

```diff
@@ -0,0 +1,39 @@
+package app.service.impl;
+
+import app.model.SupplyFruit;
+import app.service.FileWriterService;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class FileWriterServiceImplementation implements FileWriterService {
+    public static final String SEPARATOR = ",";
+    
+    @Override
+    public boolean writeData(List<SupplyFruit> fruits, String filePath) {
+        try (FileWriter writer = new FileWriter(filePath)) {
+            if (fruits.size() == 0) {
+                return false;
+            }
+            int numberOfCurrentProducts = 0;
```

Comment:
declare this variable inside of for each loop, this way you won't need to reset it's value each time in the end of the iteration